src/sinks/util/sink.rs

```diff
@@ -129,6 +133,16 @@ impl<T: Sink> Sink for StreamSink<T> {
     }
 }
 
+// === StreamSink2 ===
+
+#[async_trait]
+pub trait StreamSink2 {
+    async fn run(
+        &mut self,
+        input: Pin<Box<dyn futures::Stream<Item = Result<Event, ()>> + Send>>,
```

Comment:
How do you feel about using [`BoxStream`](https://docs.rs/futures/0.3.5/futures/stream/type.BoxStream.html) here?

Reply:
Missed this, it will be great to use `BoxStream` here!

Reply:
I recently realized this type is also useful to bind the generics.

Instead of:

```rust
Box::pin(myval) as BoxStream<_>
```

we can do

```rust
BoxStream::pin(myval)
```

I think it's pretty neat, but somehow I didn't figure it before. Just wanted to share this small tip :smile:

Reply:
Wow, I do not see this in the docs :thinking: 
`BoxStream` is `Pin<alloc::boxed::Box<dyn Stream<Item = T> + Send + 'a>>` and `Pin` do not have method `pin`, how it's possible?

Reply:
Hmm, nvm then :D